drivers/memc/memc_nxp_s32_qspi.c

```diff
@@ -1,197 +1,1707 @@
+		},))                                                                               \
+	},
+
+#define QSPI_CONTROLLER_CONFIG(n)                                                                  \
+	static uint16_t memc_nxp_qspi_rx_buffers_cfg_##n[] =                                       \
+		DT_INST_PROP_OR(n, ahb_buffers, {0});                                              \
+                                                                                                   \
+	static const struct qspi_port memc_nxp_qspi_ports_cfg_##n[] = {                            \
+		QSPI_FOREACH_PORT(n, QSPI_PORT_CONFIG)                                             \
+	};                                                                                         \
+                                                                                                   \
+	static const struct qspi_controller memc_nxp_qspi_controller_cfg_##n = {                   \
+		.ahb_base = (mem_addr_t)DT_INST_REG_ADDR_BY_NAME(n, qspi_ahb),                     \
+		.ddr = (bool)DT_INST_ENUM_IDX(n, data_rate),                                       \
+		IF_ENABLED(CONFIG_MEMC_NXP_S32_QSPI_HAS_SOCCR, (                                   \
+		.chip_options = DT_INST_PROP(n, chip_options),                                     \
+		))                                                                                 \
+		.timing = {                                                                        \
+			.aligned_2x_refclk = DT_INST_PROP(n, hold_time_2x),                        \
+			.cs = {                                                                    \
+				.hold_time = DT_INST_PROP(n, cs_hold_time),                        \
+				.setup_time = DT_INST_PROP(n, cs_setup_time),                      \
+			},                                                                         \
+			.sampling = {                                                              \
+				.delay_half_cycle = DT_INST_PROP(n, sample_delay_half_cycle),      \
+				.phase_inverted = DT_INST_PROP(n, sample_phase_inverted),          \
+			},                                                                         \
+		},                                                                                 \
+		.rx_buffers = {                                                                    \
+			.buffers = (struct qspi_buffer *)memc_nxp_qspi_rx_buffers_cfg_##n,         \
+			.buffers_cnt = (sizeof(memc_nxp_qspi_rx_buffers_cfg_##n) /                 \
+					sizeof(struct qspi_buffer)),                               \
+			.all_masters = (bool)DT_INST_PROP(n, ahb_buffers_all_masters),             \
+		},                                                                                 \
+		IF_ENABLED(CONFIG_MEMC_NXP_S32_QSPI_HAS_SFACR_CAS ||                               \
+			   CONFIG_MEMC_NXP_S32_QSPI_HAS_SFACR_BYTE_SWAP, (                         \
```

Comment:
you can not use `IF_ENABLED` with a condition that contains the operator. please update.